File: finding-seams/csharp/FindingSeams.Tests/ScheduleToTextTest.cs
using FindingSeams.ScheduleVisualising;
using Xunit;

namespace FindingSeams.Tests;

public class ScheduleToTextTest
{
    [Fact(Skip = "Exercise: finding seams — enable when working on schedule to text")]
    public async Task Prints_as_markdown()
    {
        var scheduleConverter = new ScheduleToText();
        var tsMs = DateTimeOffset.Parse("2025-11-03T00:00:00Z").ToUnixTimeMilliseconds();

        var result = await scheduleConverter.ConvertToText(
            tsMs,
            new ConvertToTextOptions { IncludeBreaks = false });

        Assert.Equal(
            """
            # Conference Schedule \u2013 Day 1 (UTC)

            * 09:00\u201310:00 Example Keynote \u2013 (Unknown Speaker) @ Main Hall [talk]
            * 10:00\u201311:30 Example workshop \u2013 (Unknown Speaker) @ TBD room [workshop]
            """ + "\n",
            result);
    }
}

File: finding-seams/kotlin/src/test/kotlin/findingseams/ScheduleToTextTest.kt
package findingseams

import findingseams.schedulevisualising.ConvertToTextOptions
import findingseams.schedulevisualising.ScheduleToText
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import java.time.Instant
import kotlin.test.assertEquals

@Disabled("Exercise: finding seams — enable when working on schedule to text")
class ScheduleToTextTest {
    @Test
    fun `prints as markdown`() = runBlocking {
        val scheduleConverter = ScheduleToText()
        val tsMs = Instant.parse("2025-11-03T00:00:00Z").toEpochMilli()

        val result = scheduleConverter.convertToText(
            tsMs,
            ConvertToTextOptions(includeBreaks = false),
        )

        assertEquals(
            """
            # Conference Schedule – Day 1 (UTC)

            * 09:00–10:00 Example Keynote – (Unknown Speaker) @ Main Hall [talk]
            * 10:00–11:30 Example workshop – (Unknown Speaker) @ TBD room [workshop]
            """.trimIndent() + "\n",
            result,
        )
    }
}
